code/modules/reagents/chemistry/equilibrium.dm

```diff
@@ -0,0 +1,226 @@
+/*
+An object/datum to contain the vars for each of the reactions currently ongoing in a holder/reagents datum
+This way all information is kept within one accessable object
+equilibrium is a unique name as reaction is already too close to chemical_reaction
+This is set up this way to reduce holder.dm bloat as well as reduce confusing list overhead
+The crux of the fermimechanics are handled here
+Instant reactions AREN'T handled here. See holder.dm
+*/
+/datum/equilibrium
+	var/datum/chemical_reaction/reaction //The chemical reaction that is presently being processed
+	var/datum/reagents/holder //The location the processing is taking place
+	var/multiplier = INFINITY
+	var/productRatio = 0
+	var/targetVol = INFINITY//The target volume the reaction is headed towards.
+	var/reactedVol = 0 //How much of the reaction has been made so far. Mostly used for subprocs
+	var/toDelete = FALSE //If we're done with this reaction so that holder can clear it
+
+/datum/equilibrium/New(datum/chemical_reaction/Cr, datum/reagents/R)
```

Comment:
Could you use better var names? Preferably in `snake_case`.

Reply:
I've changed all instances mentioned to snake_case, and I'll check over the diff for any I missed in just a moment.

Reply:
Please include `Cr` and `R` here.

Reply:
Didn't realise it it counted for here too, but I've done that as well.